go-controller/pkg/ovn/gress_policy.go

```diff
@@ -382,79 +373,45 @@ func (gp *gressPolicy) modifyACLAllow(oldMatch, newMatch string) error {
 	return nil
 }
 
-func constructIPStringForACL(direction string, ipCIDRs []string) string {
-	var v4MatchStr, v6MatchStr, matchStr string
-	v4CIDRs := make([]string, 0)
-	v6CIDRs := make([]string, 0)
-
-	for _, cidr := range ipCIDRs {
-		if utilnet.IsIPv6CIDRString(cidr) {
-			v6CIDRs = append(v6CIDRs, cidr)
+func constructIPBlockStringsForACL(direction string, ipBlocks []*knet.IPBlock) []string {
+	var matchStrings []string
+	var matchStr string
+	for _, ipBlock := range ipBlocks {
+		v4ExceptCIDRs := make([]string, 0)
+		v6ExceptCIDRs := make([]string, 0)
+		var exceptStr string
+		if utilnet.IsIPv6CIDRString(ipBlock.CIDR) {
+			matchStr = fmt.Sprintf("%s.%s == %s", "ip6", direction, ipBlock.CIDR)
 		} else {
-			v4CIDRs = append(v4CIDRs, cidr)
+			matchStr = fmt.Sprintf("%s.%s == %s", "ip4", direction, ipBlock.CIDR)
+		}
+		for _, exceptIP := range ipBlock.Except {
+			if utilnet.IsIPv6CIDRString(exceptIP) {
```

Comment:
each except IP has to be a subset of the top-level CIDR, which implies that all except IPs for a given cidr will be the same IP family as that cidr. (This is enforced by apiserver validation.) So the overall `matchStr` will be one of `ip4.DIR == CIDR`, `ip6.DIR == CIDR`, `ip4.DIR == CIDR && ip4.DIR != { EXCEPTS }` or `ip6.DIR == CIDR && ip6.DIR != { EXCEPTS }`

Reply:
did not realize that it was enforced that they had to be a subset of the CIDR block, fixed